R/pkg/R/sparkR.R

```diff
@@ -365,6 +365,10 @@ sparkR.session <- function(
     }
     overrideEnvs(sparkConfigMap, paramMap)
   }
+  if (nzchar(master)) {
+    assign("spark.master", master, envir = sparkConfigMap)
```

Comment:
I think this is not a critical issue as it is very rare to happen. We can skip it for now. And I think it is better to do in scala side (maybe throw exception in this scenario)